lib/ethers.ex

```diff
@@ -136,6 +136,23 @@ defmodule Ethers do
     end
   end
 
+    @doc """
+  Returns the native transaction (ETH) by transaction hash.
+
+  ## Parameters
+  - tx_hash: Transaction hash which the transaction is queried for.
+  - overrides:
+    - rpc_opts: Specific RPC options to specify for this request.
+  """
+  @spec get_transaction(Types.t_hash(), Keyword.t()) ::
+          {:ok, map()} | {:error, term()}
+  def get_transaction(tx_hash, opts \\ []) when is_binary(tx_hash) do
+    {rpc_client, rpc_opts} = get_rpc_client(opts)
+
+    rpc_client.eth_get_transaction_by_hash(tx_hash, rpc_opts)
+    |> post_process(nil, :get_transaction)
```

Comment:
What do you think if instead of returning whatever the JSON RPC endpoint returns, we decode the transaction to a `Ethers.Transaction` struct? That way  we can also encode it again if required!

For this we would need a `decode` function in `Ethers.Transaction` which deals with creating a new struct based on the response here.

Reply:
Thought about this but not entirely sure what is the best way here. A successful transaction has a few more fields such as `hash`, `blockHash`, `blockNumber`, `transactionIndex` etc which are not part of the transaction to be submitted. We either decorate the existing transaction struct or create a separate struct. 

Looking at [web3.py](https://github.com/ethereum/web3.py/blob/8f853f5841fd62187bce0c9f17be75627104ca43/web3/types.py#L138)
they do have all fields defined in the type. 

This will mean we need to append those missing ones to `Ethers.Transaction` too.

Reply:
That's right. Maybe we keep it simple like this for now and then decide what to do with the struct in the next major version.